### PR TITLE
Refactors cyborg spray bottles, adds cyborg weedspray to janiborgs

### PR DESCRIPTION
**Pull request by That0nePerson**
## About The Pull Request
Refactors cyborg spray bottles so that more can easily be made and they all recharge chems on their own, at the cost of power. Gives janiborg weed killer spray as a starting module, and adds an acid spray bottle to emagged janiborg. Idea by Fronsis#0357

## Why It's Good For The Game
It's much easier to add cyborg spray bottles with this, and it gives janiborg has more of an endgame use as it can fight kudzu.

## Changelog
:cl:
add: Adds weed killer spray to janiborg
add: Adds acid spray to emagged janiborg
refactor: Refactors cyborg spray bottles
/:cl:

### CODE REVIEW

**Comment by eeSPee** [comment]
Have it spray acid when emagged.

**Comment by Fronsis** [comment]
Based and thanks, how many units does the weed killer spray holds? so i can give an idea on how often i would need to go back to recharge

**Comment by That0nePerson** [comment]
> Based and thanks, how many units does the weed killer spray holds? so i can give an idea on how often i would need to go back to recharge

They recharge at 5u for 50 power and hold 250u, it shouldn't be a huge power drain.

**Comment by That0nePerson** [comment]
@eeSPee acid time

File: code/modules/reagents/reagent_containers/spray.dm [hunk 1]
```diff
@@ -357,3 +357,54 @@
...
+/obj/item/reagent_containers/spray/cyborg/process()
+	charge_tick++
+	if(charge_tick >= recharge_time)
+		regenerate_reagents()
+		charge_tick = 0
+
+/obj/item/reagent_containers/spray/cyborg/proc/regenerate_reagents()
+	if(iscyborg(src.loc))
+		var/mob/living/silicon/robot/R = src.loc
+		if(R && R.cell)
```
Comment (PowerfulBacon) [inline code/modules/reagents/reagent_containers/spray.dm]:
```suggestion
	var/mob/living/silicon/robot/R = loc
	if(istype(R))
		if(R.cell)
```

**Comment by PowerfulBacon** [comment]
Good PR, sorry I missed until now

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**